baselines/constraint/constraint_wrapper.py

```diff
@@ -70,6 +110,32 @@ def reset(self, **kwargs):
         return ob
 
     def step(self, action):
+        """
+        Perform a step in the environment and 
+        evaluate if any constraints have been violated.
+
+        Parameters
+        ----------
+        self: ConstraintEnv
+            A data structure that serves as 
+            a wrapper to the Gym environment.
+        act: ActWrapper
+            A function that takes a batch of
+            observations and returns actions.
+
+        Returns
+        -------
+        ob: Tensor?
+            An Tensor? or data structure that 
+            is augmented using the augment_obs
+            function after being reset.
+        rew: 
+            The reward provided by the environment.
+        done: Boolean?
+            Whether or not the episode is finished.
+        info: ??
```

Comment:
this is exactly what OpenAI gym returns as info, possibly with extra info added to it. You should read the docs on Gym envs.

Reply:
Thanks for the pointer, I'll go ahead and read that as well.